permissions_validators/src/public_blockchain/burn.rs

```diff
@@ -70,22 +95,33 @@ impl<W: WorldTrait> HasToken<W> for GrantedByAssetCreator {
         instruction: &Instruction,
         wsv: &WorldStateView<W>,
     ) -> Result<PermissionToken, String> {
-        let burn_box = if let Instruction::Burn(burn) = instruction {
-            burn
-        } else {
-            return Err("Instruction is not burn.".to_owned());
-        };
-        let destination_id = burn_box
-            .destination_id
-            .evaluate(wsv, &Context::new())
-            .map_err(|e| e.to_string())?;
-        let asset_id: AssetId = if let Ok(dest_id) = destination_id.try_into() {
-            dest_id
-        } else {
-            return Err("Destination is not an Asset.".to_owned());
-        };
-
-        Ok(CanBurnAssetWithDefinition::new(asset_id.definition_id).into())
+        match instruction {
+            Instruction::Unregister(unregister) => {
+                if let IdBox::AssetId(asset_id) = unregister
+                    .object_id
+                    .evaluate(wsv, &Context::new())
+                    .map_err(|e| e.to_string())?
+                {
+                    Ok(CanBurnAssetWithDefinition::new(asset_id.definition_id).into())
+                } else {
+                    Err("Instruction is not unregister asset".to_owned())
```

Comment:
here and in other similar error messages in this PR: I'd suggest saying what was expected, e.g. something like `Expected the unregister instruction` or `Expected the unregister asset instruction`